src/Uno.UI/UI/Xaml/VisualStateGroup.cs

```diff
@@ -516,6 +516,11 @@ static bool Match(VisualTransition transition, string from, string to)
 
 		internal void RefreshStateTriggers(bool force = false)
 		{
+			if (!HasStateTriggers())
+			{
+				return;
+			}
+
```

Comment:
I'm wondering why an unnecessary refresh causes GoToState to function incorrectly?